### PR TITLE
[tests] Fix pylint errors in `tests/unit`

### PR DESCRIPTION
**Pull request by cidrblock**
Related #643 

- does what it says
- several doc string/sentence fixes when seen

### CODE REVIEW

File: tests/unit/configuration_subsystem/conftest.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def _generate_config(params=None, setting_file_name=None, initial=True) -> Gener
 
     if setting_file_name:
         settings_file_path = os.path.join(TEST_FIXTURE_DIR, setting_file_name)
-        with open(settings_file_path) as file:
+        with open(file=settings_file_path, encoding="utf-8") as file:
```
Comment (webknjaz) [inline tests/unit/configuration_subsystem/conftest.py:45]: Why was this necessary? It's the default. Also, it's best to follow what the `tmp_path` does and operate on the pathlib objects. They also have `pathlib.Path.open()` if that's really necessary.

Comment (webknjaz) [inline tests/unit/configuration_subsystem/conftest.py:45]: Why was this necessary? It's the default. Also, it's best to follow what the `tmp_path` does and operate on the pathlib objects. They also have `pathlib.Path.open()` if that's really necessary.

Comment (cidrblock) [reply tests/unit/configuration_subsystem/conftest.py:45]: pylint W1514 refers to PEP 597

Comment (cidrblock) [reply tests/unit/configuration_subsystem/conftest.py:45]: @webknjaz ^

Comment (cidrblock) [reply tests/unit/configuration_subsystem/conftest.py:45]: @webknjaz see above

**Comment by cidrblock** [comment]
recheck

**Comment by cidrblock** [comment]
recheck

File: tests/unit/configuration_subsystem/conftest.py (hunk 1 above)
Comment (webknjaz) [reply tests/unit/configuration_subsystem/conftest.py:45]: > pylint W1514 refers to PEP 597

Oh, interesting, I didn't realize. This raises another question — why didn't pytest reveal the warning. Ah! It's probably because we don't test on Python 3.10 (and maybe pytest also isn't set up right...)

But this is resolved for now, not really related to the PR.

Comment (webknjaz) [reply tests/unit/configuration_subsystem/conftest.py:45]: Oh, it's a separate question but anyway it's not the right place to solve it. Resolved.

**Review by webknjaz — APPROVED** [review]
